compiler/src/iree/compiler/InputConversion/StableHLO/LegalizeToLinalgUtils.cpp

```diff
@@ -0,0 +1,133 @@
+// Copyright 2022 The TensorFlow Authors. All Rights Reserved.
+// Copyright 2023 The IREE Authors
```

Comment:
Given that we're relicensing, we want to drop the TF Copyright line as well. The [Google CLA](https://cla.developers.google.com/about), which all contributors to this have signed, permits this sort of relicensing. As a courtesy, we've also reached out to major contributors to these files to make sure they don't have any specific concerns. I think it would be good to explicitly note in the description that we are relicensing this code under the nearly-identical Apache-2.0 WITH LLVM-exception license.

Copyright continues to be maintained by whoever actually wrote the code (or their employer if they did under the auspices of their employer). If we look at the blame of these files, I think all the authors are covered under IREE's AUTHORS file already. Please confirm that. If not, we can add them. We could actually leave the original year here (from the TF side), I think.

If we care about preserving the records of authorship, I think the more useful thing to do would actually be to preserve the git history of the files and use a merge commit to bring them into our repo. That would involve using git filter-branch or git filter-tree to narrow it to just the commits for these files. It depends a bit on how messy the history is though (TF has some not great git hygiene) and how much work it is, but it would be nice to take a look.

Reply:
Hmm, AFAIK this is how stablehlo preserved their ties to mhlo, e.g.: https://github.com/openxla/stablehlo/blob/7f6976e15595d3a37ad177a58b4a3c2024d9ae9c/stablehlo/dialect/StablehloOps.td.

I will look into the other options you mentioned; I really appreciate the level of detail of this suggestion.

Reply:
We've relicensed things before, updating the authors copyright block (after checking with oss-licensing that this was OK). I don't think it's useful to preserve the different name, since it's basically an alias for a list of people and if any people are significant contributors to the files here and aren't in our list then we should add them, since this is now part of IREE and they are therefore one of the authors :-)

@burmako in case there was a particular reason you did it this way previously

Reply:
Agreed on not preserving (large part was IREE authors already, TF's authors [differs](https://github.com/tensorflow/tensorflow/blob/master/AUTHORS) but if we look at code here I don't think any covered by TF's is not covered by IREE's). TF's git history is rather messy and folks can already use the git rev above, go to the other repo and get full history & blame, but good to check if we got lucky with these files.

Reply:
Done. I kept only the 'The IREE Authors' line in every header. I'll update the PR description to make the relicensing part very prominent.

When it comes to other authors of these files, I found that PolyMage Labs and ByteDance made one commit each, but because these commits were <100 LoC I didn't update the AUTHORS file.

I looked into preserving the file history, but this seems messy because I will import more files in future PR, so this 'history import' would have to be done multiple times each time we pull a new file in.